test/user_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+  def setup
+    @sal = User.new("Sal")
+    @ali = User.new("Ali")
+    @joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    @joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+  end
+
+  def test_it_exists
+    assert_instance_of User, @sal
+  end
+
+  def test_it_has_attributes
+    assert_equal "Sal", @sal.name
+    assert_equal "Ali", @ali.name
+    assert_equal [], @sal.jokes
+  end
+
+  def test_sal_has_jokes
+    @sal.learn(@joke_1)
+    @sal.learn(@joke_2)
+
+    assert_equal [@joke_1, @joke_2], @sal.jokes
+  end
+
+  def test_ali_has_jokes
+    @sal.tell(@ali, @joke_1)
+    @sal.tell(@ali, @joke_2)
+
+    assert_equal [@joke_1, @joke_2], @ali.jokes
```

Comment:
Good start here. Sal shouldn't be able to tell jokes that he hasn't learned himself, though, so you would need to copy the `@sal.learn()` methods from lines 25 and 26 here.